modules/navier_stokes/src/fvkernels/INSFVMomentumAdvection.C

```diff
@@ -49,8 +59,27 @@ INSFVMomentumAdvection::INSFVMomentumAdvection(const InputParameters & params)
     INSFVMomentumResidualObject(*this),
     _rho(getFunctor<ADReal>(NS::density)),
     _approximate_as(isParamValid("characteristic_speed")),
-    _cs(_approximate_as ? getParam<Real>("characteristic_speed") : 0)
+    _cs(_approximate_as ? getParam<Real>("characteristic_speed") : 0),
+    _dim(_subproblem.mesh().dimension()),
+    _u(getFunctor<ADReal>(NS::velocity_x)),
+    _v(getFunctor<ADReal>(NS::velocity_y)),
+    _w(getFunctor<ADReal>(NS::velocity_z)),
+    _absolute_momentum_limiter(getParam<bool>("momentum_norm_limiter"))
 {
+  if (_absolute_momentum_limiter)
+  {
+    if (!isParamSetByUser(NS::velocity_x))
+      paramError(NS::velocity_x,
+                 "The u velocity must be defined when using the absolute value momentum limiter.");
+    if (_dim >= 2 && !isParamSetByUser(NS::velocity_y))
+      paramError(NS::velocity_y,
+                 "The v velocity must be defined when using the absolute value momentum limiter"
+                 "and problem dimension is larger or equal to 2.");
+    if (_dim >= 3 && !isParamSetByUser(NS::velocity_z))
+      paramError(NS::velocity_z,
+                 "The w velocity must be defined when using the absolute value momentum limiter"
```

Comment:
Just checking that u/v/w nomenclature is something that users understand

Reply:
They do